gluten-core/src/main/scala/io/glutenproject/GlutenPlugin.scala

```diff
@@ -64,6 +64,13 @@ private[glutenproject] class GlutenDriverPlugin extends DriverPlugin {
       s"${GLUTEN_SESSION_EXTENSION_NAME}"
     }
     conf.set(SPARK_SESSION_EXTS_KEY, String.format("%s", extensions))
+    // FIXME Hongze 22/12/06
+    //  BatchScan.scala in shim was not always loaded by class loader.
+    //  The file should be removed and the "ClassCastException" issue caused by
+    //  spark.sql.parquet.enableVectorizedReader=true should be fixed in another way.
+    //  Before the issue was fixed we force the use of vanilla row reader by using
+    //  the following statement.
+    conf.set("spark.sql.parquet.enableVectorizedReader", "false")
```

Comment:
I remember this config can only take effect for datasource v1, and that's why we copied BatchScanExec into gluten. Maybe some notes can be added here if that's the case.

Reply:
I double checked and the options seem to be picked both in v1 and v2

v1
https://github.com/apache/spark/blob/97976a5cc915597fd2606602d18c52c075a03bf6/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala#L168-L169

v2
https://github.com/apache/spark/blob/97976a5cc915597fd2606602d18c52c075a03bf6/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/parquet/ParquetPartitionReaderFactory.scala#L74

Reply:
Thanks for the check!